src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/WarningLevelOverriddenValueProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties;
+
+/// <summary>
+/// Provides a synthetic property indicating if the <c>WarningLevel</c> set by the
+/// user (if any) will be overridden by the SDK.
+/// </summary>
+/// <remarks>
+/// Based on the logic in https://github.com/dotnet/sdk/blob/a35fecf11708d1de6eb86f3dc3294f1644d3212e/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets.
+/// To summarize: in the .NET 5.0 SDK and up, the <c>EffectiveAnalysisLevel</c>
+/// property is computed from the <c>AnalysisLevel</c> property. If
+/// <c>EffectiveAnalysisLevel</c> is >= 5.0, the SDK will override any previously-set
+/// value of <c>WarningLevel</c> with its own.
```

Comment:
There is a downside to coupling the implementation policy here to that in the SDK. In future, as the SDK changes, it can create issues when pinning the SDK.

Ideally the SDK would expose this information to the project system, as is done for target frameworks today. Older SDKs wouldn't provide that info, so we'd need some fallback.